lib/WP_Auth0_LoginManager.php

```diff
@@ -163,136 +170,124 @@ public function init_auth0() {
       } else {
         $this->redirect_login();
       }
-    } catch (WP_Auth0_LoginFlowValidationException $e) {
+    } catch ( WP_Auth0_LoginFlowValidationException $e ) {
 
+      // Errors during the OAuth login flow
       $this->die_on_login( $e->getMessage(), $e->getCode() );
 
-    } catch (WP_Auth0_BeforeLoginException $e) {
+    } catch ( WP_Auth0_BeforeLoginException $e ) {
 
+      // Errors during the WordPress login flow
       $this->die_on_login( $e->getMessage(), $e->getCode(), FALSE );
-
-    } catch (Exception $e) {
-
     }
-
   }
 
+  /**
+   * Main login flow, Authorization Code Grant
+   *
+   * @throws WP_Auth0_BeforeLoginException
+   * @throws WP_Auth0_LoginFlowValidationException
+   *
+   * @see https://auth0.com/docs/api-auth/tutorials/authorization-code-grant
+   */
   public function redirect_login() {
-    global $wp_query;
-
-    if ( $this->query_vars( 'auth0' ) === null ) {
```

Comment:
These 3 `if` blocks were moved to `init_auth0()` to be caught earlier and for both redirect and implicit flows.